pkg/operator/reports_test.go

```diff
@@ -92,3 +101,154 @@ func TestGetNextReportPeriod(t *testing.T) {
 		})
 	}
 }
+
+func TestIsReportRunning(t *testing.T) {
+	const (
+		testNamespace  = "default"
+		testReportName = "test-report"
+		testQueryName  = "test-query"
+	)
+
+	schedule := &metering.ReportSchedule{
+		Period: metering.ReportPeriodCron,
+		Cron:   &metering.ReportScheduleCron{Expression: "5 4 * * *"},
+	}
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name       string
+		report     *metering.Report
+		expectFlag bool
+	}{
+		{
+			name:       "new report returns false",
+			report:     testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectFlag: false},
+		{
+			name: "run-once report returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, nil, false),
+			expectFlag: true},
+		{
+			name: "reportingEnd is unset returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, nil, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFlag: false},
+		{
+			name: "reportingEnd > lastReportTime returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 0, 0)},
+			}, schedule, false),
+			expectFlag: false},
+		{
+			name: "reportingEnd < lastReportTime returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+			}, schedule, false),
+			expectFlag: true},
+	}
```

Comment:
You could add some more test cases where the reason isn't finished just to cover the bases.